### ISSUE TITLE
missing global for short-circuit assignment

### ISSUE THREAD

**Issue by stevengj** [stevengj/SoftGlobalScope.jl#18]
The following gives `UndefVarError: s not defined`:
```jl
s = 0
for i = 1:10
    true && (s += i)
end
s
```
(Reported by @zhangliye [on discourse](https://discourse.julialang.org/t/simple-is-better-than-complex/18821/4?u=stevengj).)

**Comment by stevengj** [comment]
Looks like the `_softscope` function just needs to handle `&&` and `||` nodes, as well as `comparison` nodes.  The following case also fails:
```jl
s = 0
for i = 1:10
    10 > (s += i) > 0
end
s
```

**Comment by stevengj** [comment]
Fixed in version 1.0.8 (JuliaLang/METADATA.jl#20174).
